### ISSUE TITLE
Data conversion error with BZERO on input

### ISSUE THREAD

**Issue by elwinter** [astroconda/astroconda#28]
Hey folks,

We have a problem with the code in the latest astroconda for reading compressed FITS files. Specifically, the following commands reproduce the error:

...
source activate astroconda
python
import astropy.io.fits as pyfits
d, h = pyfits.getdata("/ifs/cs/projects/armin1/data/v20.0/PS1/GPC1v3/rawdata/F15akt/tmpl/g/c",header=True)
...

The resulting error message is:

...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/user/ewinter/anaconda2/envs/astroconda/lib/python2.7/site-packages/astropy/io/fits/convenience.py", line 200, in getdata
    data = hdu.data
  File "/user/ewinter/anaconda2/envs/astroconda/lib/python2.7/site-packages/astropy/utils/decorators.py", line 498, in **get**
    val = self.fget(obj)
  File "/user/ewinter/anaconda2/envs/astroconda/lib/python2.7/site-packages/astropy/io/fits/hdu/compressed.py", line 1408, in data
    data += self._bzero
TypeError: Cannot cast ufunc add output from dtype('float64') to dtype('uint16') with casting rule 'same_kind'
...

Examining the code in compressed.py in the debugger shows the problem - the self._bzero is type float64, while the array data is type uint16, and Python will not automatically cast down from the former to the latter. This astroconda code (dated 22 June 2016) appears to be essentially unchanged from an earlier version that was part of the Ureka distribution (dated around 13 May 2014), other than an added return statement when data is None. The earlier, Ureka version of compressed.py successfully reads the file.

So put simply, some old Ureka code can read a compressed FITS file, but the new code in astroconda cannot. I can provide the FITS file if it is needed for investigation.

Thanks
Eric Winter

**Comment by pllim** [comment]
Looks like this has been reported at astropy/astropy#4588. AstroConda has no control over bugs in the packages it distributes. I don't think this is the right place for this issue.

**Comment by elwinter** [comment]
OK, fair enough. I hadn’t checked the astropy list. Thanks for the follow-up.

Eric
